server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/DefaultProject.java

```diff
@@ -82,6 +89,14 @@ public class DefaultProject implements Project {
         boolean success = false;
         try {
             createReservedRepos(System.currentTimeMillis());
+            final UserAndTimestamp creation = metadataCreation();
+            if (creation != null) {
```

Comment:
nit; personally I prefer that the branching is done on `REPO_DOGMA` for clarity, but I guess functionally the same is achieved in this block. No problem to leave as-is.

```
    if (REPO_DOGMA.equals(name)) {
        creationTimeMillis = repos.get(REPO_DOGMA).creationTimeMillis();
        author = repos.get(REPO_DOGMA).author();
    } else {
        final UserAndTimestamp creation = metadataCreation(); // fallbacks to Author.SYSTEM
        creationTimeMillis = creation.epochMillis();
        author = Author.ofEmail(creation.user());
    }
```

Reply:
I've used the current logic to handle the case when creating metadata fails. (I'm not sure if there's a chance for that event to occur though.)
https://github.com/line/centraldogma/blob/81958a67ac630ee5b306a0229f5b73ce27e6e5d6/server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/DefaultProject.java#L149-L150
If it fallbacks to Author.SYSTEM in that case, it has to create `Author` again event if it can get it from `repos.get(REPO_DOGMA).author()`. So let me leave it as is. 😉